**Ya estamos en condiciones de empezar a generar insights de cara al MVP nuevamente.**

¿Con qué se encuentra alguien que quiere elegir una moto?

- Una inmensa variedad: Miles de modelos con nombres y características similares pero diferencias sutiles.

- Jergas técnicas: Mucha terminología (V-Twin, par motor, DCT, etc.) que puede abrumar.

- Necesidades vs. Deseos: A menudo, lo que creen que quieren no se ajusta a lo que necesitan realmente para su uso diario o experiencia.

- Presupuesto: El factor limitante número uno.

- Percepción de marca: La fama, el prestigio o la reputación de una marca.

- Factores emocionales: El diseño, el sonido, la "sensación" al subirse.

Qué podemos considerar y utilizar, de cara a un comparador o recomendador.

1. Datos Esenciales
- MARCA: Crucial para la fidelidad a la marca o para buscar alternativas.
- MODELO: Identificador único de cada moto.
- TIPO DE MOTO: Quiero una naked, una trail, una scooter...

- Cilindrada (cc): Importante. Define directamente la potencia, el tipo de carnet necesario (A1, A2, A) y el segmento de la moto.
¿Por qué es importante?: Restricciones de carnet, tipo de uso (urbano, carretera, viajes), coste del seguro, consumo.

- Potencia (CV o kW): Ligado a la cilindrada, pero no lineal.
¿Por qué es importante?: Sensaciones de conducción, aceleración, velocidad máxima, aptitud para viajar con pasajero o carga.

- Tipo de Motor: (Ej: Monocilíndrico, Bicilíndrico en línea, V-Twin, Cuatro cilindros en línea, Bóxer). 
Cada configuración tiene un sonido, una entrega de potencia y unas vibraciones características.
ES UN DATO QUE NO TENEMOS DISPONIBLE POR AHORA

- Tipo de Transmisión: (Manual, DCT, Automática para scooters).
¿Por qué es importante?: Facilidad de conducción (especialmente DCT/automático en ciudad), experiencia de conducción.
NO LO TENEMOS DIRECTAMENTE, EN ALGUNOS CASOS SE PUEDE INFERIR DESDE EL TIPO

- Peso (kg): Influye enormemente en la manejabilidad, especialmente en parado o a baja velocidad.
¿Por qué es importante?: Facilidad para principiantes, manejo en ciudad, capacidad de maniobra, facilidad de aparcamiento.

- Altura del Asiento (mm): Dato ergonómico vital para la comodidad y seguridad del piloto.
¿Por qué es importante?: Permite al piloto llegar al suelo con los pies, lo que da confianza, especialmente a personas de menor estatura o principiantes.
- Capacidad del Depósito (Litros) y Autonomía (km): Para viajes largos.
Consumo de Combustible (L/100km): Afecta directamente el coste de uso.
NO DISPONIBLE AUN


3. Datos de Equipamiento y Tecnología (aún no incoporados en esta etapa de desarrollo):
Sistema de Frenado (ABS, CBS): La seguridad es primordial. El ABS es casi un estándar hoy en día.
Electrónica Avanzada: (Control de tracción, modos de conducción, control de crucero, quickshifter, etc.). Estos sistemas mejoran la seguridad y la experiencia de conducción.
Tipo Iluminación: (LED, halógena).
Panel de Instrumentos: (Digital, TFT, analógico).
Conectividad: (Bluetooth, GPS integrado).
Equipamiento Específico: (Maletas integradas, defensas, cúpulas ajustables, puños calefactables, etc.). Esto es crucial para motos touring o adventure.

4. Datos de Precio y Mercado:

- Precio Nuevo (EUR): Absolutamente fundamental. 
El presupuesto es el primer filtro para la mayoría de los compradores.

PENDIENTES
- Precio de Reventa / Valor Usado (EUR): Aunque más complejo de obtener, es un dato que muchos compradores consideran implícitamente.

- Disponibilidad (Nuevo/Usado): ¿Se sigue fabricando? ¿Es un modelo que solo se encuentra de segunda mano?


5. Datos de la Experiencia del Usuario (Más avanzados para un recomendador, pero valiosos para el análisis):
- Puntuaciones/Valoraciones de Usuarios: Opiniones subjetivas de quienes ya poseen la moto.


- Comentarios y Reseñas: Información cualitativa sobre problemas comunes, virtudes inesperadas, etc.

- Fiabilidad General: Percepción de la marca/modelo en cuanto a averías.


🚀 ¿Cómo elegir tu primera (o próxima) moto sin volverte loco?
En MotoFit analizamos +500 modelos para ayudarte a comparar y decidir mejor.

In [1]:
import pandas as pd

df = pd.read_csv('motofit_v2.csv')
df = df.drop(columns=['Unnamed: 0'])

df.head(10)

MODELO CON_CARNET CILINDRADA POTENCIA RUEDA_DELANTERA  \
0                  SXR         AM         49      3.2       120/70-12   
1               RX 125         A1        124     14.9        90/90-21   
2               SX 125         A1        124     14.9        90/90-21   
3               RS 125         A1        124     14.9       100/80-17   
4            Tuono 125         A1        124     14.9       100/80-17   
5            SR GT 125         A1        124     14.9       110/80-14   
6        SR GT ABS 125         A1        124     14.9       110/80-14   
7      SR GT Sport 125         A1        124     14.9       110/80-14   
8  SR GT Sport ABS 125         A1        124     14.9       110/80-14   
9               RS 457         A2        457     48.0      110/70ZR17   

  RUEDA_TRASERA FRENO_DELANTERO FRENO_TRASERO ALTURA_ASIENTO PESO_VACIO  \
0     120/70-12            D220          D220            775       n.d.   
1     120/80-18            D260          D220            880        134   
2     120/80-18            D260          D220            880        134   
3    140/70ZR17            D300          D218            820        144   
4    140/70ZR17            D300          D218            799        144   
5     130/70-13            D260          D220            799        148   
6     130/70-13            D260          D220            799        150   
7     130/70-13            D260          D220            799        148   
8     130/70-13            D260          D220            799        150   
9    150/60ZR17           2D320          D220            820        175   

   PRECIO    MARCA CARNET_MINIMO TIPO_CORREGIDO TIPO_SIMPLIFICADO  
0  2599.0  APRILIA            AM        Scooter           Scooter  
1  4199.0  APRILIA            A1       Off-road          Off-road  
2  4199.0  APRILIA            A1       Off-road          Off-road  
3  5499.0  APRILIA            A1          Sport             Sport  
4  5399.0  APRILIA            A1          Naked             Naked  
5  4099.0  APRILIA            A1        Scooter           Scooter  
6  4199.0  APRILIA            A1        Scooter           Scooter  
7  4099.0  APRILIA            A1        Scooter           Scooter  
8  4199.0  APRILIA            A1        Scooter           Scooter  
9  7199.0  APRILIA            A2          Sport             Sport

In [2]:
# Nos aseguramos que las columnas numéricas estén en formato adecuado
cols_numericas = ['POTENCIA', 'PESO_VACIO', 'ALTURA_ASIENTO', 'CILINDRADA', 'PRECIO']
for col in cols_numericas:
    df[col] = pd.to_numeric(df[col], errors='coerce') 

# Creamos tabla dinámica agrupando por tipo simplificado
tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'  # promedio por tipo
).round(1)

# Mostrar la tabla
print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)

=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      737.8      1101.3       254.9      70.4  16556.2
Naked                       797.5       644.3       460.5      72.8   9722.0
Off-road                    771.6      1217.9       290.0      44.4  25094.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.5       187.8       153.4      18.0   5161.4
Sport                       832.8       727.4      1042.1     116.8  17611.9
Tourer                      776.2      1197.3       285.9      99.9  22172.9


In [3]:
# Me llama la atencion el alto promedio de cilindrada en custom y en off-road
# Filtrar todas las motos clasificadas como Custom
motos_custom = df[df['TIPO_SIMPLIFICADO'] == 'Custom']

# Mostrar el resultado
print(motos_custom)



                MODELO CON_CARNET  CILINDRADA  POTENCIA RUEDA_DELANTERA  \
30               502 C         A2       500.0      47.8     120/70 ZR17   
61   R nineT Scrambler          A      1170.0     110.0      120/70 R19   
62   R nineT Urban G/S          A      1170.0     110.0      120/70 R19   
75                R 18          A      1802.0      91.0      120/70 R19   
76              R 18 B          A      1802.0      91.0      120/70 R19   
..                 ...        ...         ...       ...             ...   
510  Rocket 3 Storm GT          A      2548.0     167.0      150/80 R17   
578             G1-125         A1       125.0      14.7      110/70-R17   
579           G1-125 X         A1       125.0      14.7      110/70-R17   
580               125C         A1       125.0      14.7      110/70-R17   
593               350V         A2       348.0      39.4      120/70-R17   

    RUEDA_TRASERA FRENO_DELANTERO FRENO_TRASERO  ALTURA_ASIENTO  PESO_VACIO  \
30    160/60 ZR17   

In [4]:
motos_off = df[df['TIPO_SIMPLIFICADO'] == 'Off-road']
print(motos_off)

                    MODELO CON_CARNET  CILINDRADA  POTENCIA RUEDA_DELANTERA  \
1                   RX 125         A1       124.0      14.9        90/90-21   
2                   SX 125         A1       124.0      14.9        90/90-21   
213             701 Enduro          A       692.0      74.0        90/90-21   
238             Roadmaster          A      1890.0       NaN      130/80 B17   
239  Roadmaster Dark Horse          A      1890.0       NaN      130/60 B19   
240     Roadmaster Limited          A      1890.0       NaN      130/60 B19   
241        Pursuit Limited          A      1768.0       NaN      130/60 B19   
243       Roadmaster Elite          A      1890.0       NaN      130/60 B19   
303           690 Enduro R          A       692.7      74.0        90/90-21   

    RUEDA_TRASERA FRENO_DELANTERO FRENO_TRASERO  ALTURA_ASIENTO  PESO_VACIO  \
1       120/80-18            D260          D220           880.0       134.0   
2       120/80-18            D260          D220    

In [5]:
#en off-road claramente tenemos un error con las indians
df.loc[df['MODELO'].str.contains("Roadmaster|Pursuit", case=False), 'TIPO_SIMPLIFICADO'] = 'Tourer'


In [6]:
motos_off = df[df['TIPO_SIMPLIFICADO'] == 'Off-road']
print(motos_off)

           MODELO CON_CARNET  CILINDRADA  POTENCIA RUEDA_DELANTERA  \
1          RX 125         A1       124.0      14.9        90/90-21   
2          SX 125         A1       124.0      14.9        90/90-21   
213    701 Enduro          A       692.0      74.0        90/90-21   
303  690 Enduro R          A       692.7      74.0        90/90-21   

    RUEDA_TRASERA FRENO_DELANTERO FRENO_TRASERO  ALTURA_ASIENTO  PESO_VACIO  \
1       120/80-18            D260          D220           880.0       134.0   
2       120/80-18            D260          D220           880.0       134.0   
213     140/80-18            D320          D240           910.0       145.0   
303     140/80-18            D300          D240           910.0       146.0   

      PRECIO      MARCA CARNET_MINIMO TIPO_CORREGIDO TIPO_SIMPLIFICADO  
1     4199.0    APRILIA            A1       Off-road          Off-road  
2     4199.0    APRILIA            A1       Off-road          Off-road  
213  12249.0  HUSQVARNA           

In [7]:
#VEAMOS NUEVAMENTE
# Nos aseguramos que las columnas numéricas estén en formato adecuado
cols_numericas = ['POTENCIA', 'PESO_VACIO', 'ALTURA_ASIENTO', 'CILINDRADA', 'PRECIO']
for col in cols_numericas:
    df[col] = pd.to_numeric(df[col], errors='coerce') 

# Creamos tabla dinámica agrupando por tipo simplificado
tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'  # promedio por tipo
).round(1)

# Mostrar la tabla
print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)

=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      737.8      1101.3       254.9      70.4  16556.2
Naked                       797.5       644.3       460.5      72.8   9722.0
Off-road                    895.0       408.2       139.8      44.4   8224.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.5       187.8       153.4      18.0   5161.4
Sport                       832.8       727.4      1042.1     116.8  17611.9
Tourer                      763.6      1278.8       301.0      99.9  24175.0


In [8]:

tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'
).round(1)


print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)

=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      737.8      1101.3       254.9      70.4  16556.2
Naked                       797.5       644.3       460.5      72.8   9722.0
Off-road                    895.0       408.2       139.8      44.4   8224.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.5       187.8       153.4      18.0   5161.4
Sport                       832.8       727.4      1042.1     116.8  17611.9
Tourer                      763.6      1278.8       301.0      99.9  24175.0


In [9]:
# Ver todas las motos clasificadas como Sport
sport_motos = df[df['TIPO_SIMPLIFICADO'] == 'Sport'][['MARCA', 'MODELO', 'PESO_VACIO']]

# Filtrar las que parecen tener errores (peso exagerado)
errores_peso = sport_motos[sport_motos['PESO_VACIO'] > 400]

print(errores_peso)


      MARCA                    MODELO  PESO_VACIO
111  DUCATI               Panigale V2     17990.0
112  DUCATI             Panigale V2 S     17990.0
113  DUCATI  Panigale V2 SQ Final Ed.     19980.0


Acá seguramente está el precio en la columna equivocada. Revisando la data original. Los valores originales corresponden a 179, 176, 190. y los precios respectivos: 17490, 19890, 30500

In [10]:
# Corregir manualmente los errores de peso
df.loc[df['MODELO'] == 'Panigale V2', 'PESO_VACIO'] = 179.0
df.loc[df['MODELO'] == 'Panigale V2 S', 'PESO_VACIO'] = 176.0
df.loc[df['MODELO'] == 'Panigale V2 SQ Final Ed.', 'PESO_VACIO'] = 190.0

# Corregir precios también, si están mal
df.loc[df['MODELO'] == 'Panigale V2', 'PRECIO'] = 17490
df.loc[df['MODELO'] == 'Panigale V2 S', 'PRECIO'] = 19890
df.loc[df['MODELO'] == 'Panigale V2 SQ Final Ed.', 'PRECIO'] = 30500


In [11]:
tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'
).round(1)


print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)

=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      737.8      1101.3       254.9      70.4  16556.2
Naked                       797.5       644.3       460.5      72.8   9722.0
Off-road                    895.0       408.2       139.8      44.4   8224.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.5       187.8       153.4      18.0   5161.4
Sport                       832.8       727.4       189.6     116.8  17611.9
Tourer                      763.6      1278.8       301.0      99.9  24175.0


In [12]:
df[df['MARCA'] == 'DUCATI'][['MODELO','PRECIO']]


MODELO   PRECIO
99                     Monster  12990.0
100                   Monster+  13490.0
101                 Monster SP  16090.0
102             Supersport 950  15890.0
103                    DesertX  17890.0
104              DesertX Rally  23290.0
105             Multistrada V2  16090.0
106           Multistrada V2 S  19290.0
107            Hypermotard 698  13690.0
108            Hypermotard 950  16590.0
109         Hypermotard 950 SP  20190.0
110        Hypermotard 950 RVE  17590.0
111                Panigale V2  17490.0
112              Panigale V2 S  19890.0
113   Panigale V2 SQ Final Ed.  30500.0
114           Streetfighter V2  16590.0
115         Streetfighter V2 S  18990.0
116                  Diavel V4  30990.0
117             Multistrada V4  22790.0
118       Multistrada V4 Rally  30790.0
119           Multistrada V4 S  27190.0
120     Multistrada V4 S Sport  31300.0
121          Multistrada V4 RS  42190.0
122  Multistrada V4 Pikes Peak  34690.0
123           Streetfighter V4  26126.0
124         Streetfighter V4 S  30126.0
125       Streetfighter V4 SP2  38690.0
126                Panigale V4  31590.0
127              Panigale V4 S  37990.0
128                       Icon  11690.0
129                  Icon Dark  10690.0
130              Full Throttle  12990.0
131                 Nightshift  12990.0
132        10 Anniv. Rizoma Ed  15190.0
133             1100 Sport Pro  17390.0

mejor. ahora Ducati se correlaciona con la data original. Seguiremos con el peso de las motos Naked.
Un promedio de más de 400 kilos no corresponde. 

In [13]:
# Ver todas las motos clasificadas como Sport
naked_motos = df[df['TIPO_SIMPLIFICADO'] == 'Naked'][['MARCA', 'MODELO', 'PESO_VACIO']]

# Filtrar las que parecen tener errores (peso exagerado)
errores_peso = naked_motos[naked_motos['PESO_VACIO'] > 300]

print(errores_peso)

      MARCA              MODELO  PESO_VACIO
114  DUCATI    Streetfighter V2     19090.0
115  DUCATI  Streetfighter V2 S     17590.0
346    MITT          Legend 301      3395.0
350    MITT       808 Traveller       332.0


In [14]:
# Corregir manualmente los errores de peso
df.loc[df['MODELO'] == 'Streetfighter V2', 'PESO_VACIO'] = 178
df.loc[df['MODELO'] == 'Streetfighter V2 S', 'PESO_VACIO'] = 175
df.loc[df['MODELO'] == 'Legend 301', 'PESO_VACIO'] = 162

# Corregir precios también, si están mal
df.loc[df['MODELO'] == 'Streetfighter V2', 'PRECIO'] = 16590
df.loc[df['MODELO'] == 'Streetfighter V2 S', 'PRECIO'] = 18990
df.loc[df['MODELO'] == 'Legend 301', 'PRECIO'] = 3395

In [16]:
tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'
).round(1)


print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)


=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      737.8      1101.3       254.9      70.4  16556.2
Naked                       797.5       644.3       185.8      72.8   9722.0
Off-road                    895.0       408.2       139.8      44.4   8224.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.5       187.8       153.4      18.0   5161.4
Sport                       832.8       727.4       189.6     116.8  17611.9
Tourer                      763.6      1278.8       301.0      99.9  24175.0


In [23]:
precio_scooter = df[df['TIPO_SIMPLIFICADO'] == 'Scooter'][['MARCA', 'MODELO', 'CILINDRADA','PRECIO']]
precio_scooter.head(10)

MARCA               MODELO  CILINDRADA   PRECIO
0            APRILIA                  SXR        49.0   2599.0
5            APRILIA            SR GT 125       124.0   4099.0
6            APRILIA        SR GT ABS 125       124.0   4199.0
7            APRILIA      SR GT Sport 125       124.0   4099.0
8            APRILIA  SR GT Sport ABS 125       124.0   4199.0
41               BMW              C 400 X       350.0   7250.0
42               BMW             C 400 GT       350.0   8290.0
43               BMW                CE 02         NaN   8425.0
44               BMW                CE 04         NaN  12075.0
160  HARLEY DAVIDSON         LiveWire One        15.5  21759.0

Las "scooter" fuera de rango, no es scooter es electrica. Las eléctricas están sin representación específica.
Me parece que lo mejor será mantener la categorización que les corresponde de acuerdo a su parte ciclo y crear una manera de categorizar la propulsión de todas las motocicletas. 
Muy probablemente las que no tienen cilindrada sean electricas, vamos a chequear eso. 
Hay una Harley que ES elecrica con cilidrada, chequearemos esa situaicón luego.

In [22]:

df[df['CILINDRADA'].isnull()][['MARCA', 'MODELO', 'TIPO_SIMPLIFICADO']]


MARCA                     MODELO TIPO_SIMPLIFICADO
43        BMW                      CE 02           Scooter
44        BMW                      CE 04           Scooter
163     HONDA                      EM1 e           Scooter
245  KAWASAKI                  Ninja e-1             Sport
262  KAWASAKI                      Z e-1             Naked
373   PEUGEOT            E-Streetzone L1           Scooter
399   PIAGGIO      Piaggio 1 electric 45           Scooter
400   PIAGGIO    Piaggio 1 + electric 45           Scooter
401   PIAGGIO  Piaggio 1 Activ. elec. 60           Scooter
527     VESPA     Sprint Ellectrica 45 S           Scooter
528     VESPA    Primavera Ellectrica 45           Scooter
529     VESPA   Primavera Elett. 70 Tech           Scooter
534      VOGE                       ER10             Naked

In [ ]:
# Efectivamente existen motos eléctricas que tienen asignada una cilindrada en nuestro df, ej. Harleys.
df[df['MODELO'].str.contains("LiveWire", case=False)][['MODELO', 'CILINDRADA', 'POTENCIA', 'PESO_VACIO', 'PRECIO']]


MODELO  CILINDRADA  POTENCIA  PESO_VACIO   PRECIO
160  LiveWire One        15.5     105.0       255.0  21759.0

In [25]:
# Empezaremos con catalogar de eléctricas las que seguramente lo son (cilindrada NaN que acabamos de ver), 
# luego buscaremos las "escondidas", como las harley por ejemplo.


df['CILINDRADA'] = pd.to_numeric(df['CILINDRADA'], errors='coerce')
df.loc[df['CILINDRADA'].isnull(), 'CILINDRADA'] = 0

# Crear nueva columna de propulsión (combustión por defecto)
df['PROPULSION'] = 'Combustión'

# Asignar 'Eléctrica' a las que tienen cilindrada 0
df.loc[df['CILINDRADA'] == 0, 'PROPULSION'] = 'Eléctrica'

print("Cantidad de motos eléctricas detectadas:", (df['PROPULSION'] == 'Eléctrica').sum())


Cantidad de motos eléctricas detectadas: 13


In [ ]:
# ahora buscaremos las de cilindrada extrañamente baja.
df_cil_baja = df[df['CILINDRADA'] < 30][['MARCA', 'MODELO','CILINDRADA']]
df_cil_baja.head(20)

MARCA                     MODELO  CILINDRADA
43               BMW                      CE 02         0.0
44               BMW                      CE 04         0.0
160  HARLEY DAVIDSON               LiveWire One        15.5
161  HARLEY DAVIDSON                 S2 Del Mar        10.5
162  HARLEY DAVIDSON              S2 Mulholland        10.5
163            HONDA                      EM1 e         0.0
245         KAWASAKI                  Ninja e-1         0.0
262         KAWASAKI                      Z e-1         0.0
373          PEUGEOT            E-Streetzone L1         0.0
399          PIAGGIO      Piaggio 1 electric 45         0.0
400          PIAGGIO    Piaggio 1 + electric 45         0.0
401          PIAGGIO  Piaggio 1 Activ. elec. 60         0.0
527            VESPA     Sprint Ellectrica 45 S         0.0
528            VESPA    Primavera Ellectrica 45         0.0
529            VESPA   Primavera Elett. 70 Tech         0.0
534             VOGE                       ER10         0.0

In [29]:
# Asignar cilindrada 0 a motos con cilindrada menor a 16 cc (posibles eléctricas mal ingresadas)
df.loc[df['CILINDRADA'] < 16, 'CILINDRADA'] = 0


In [30]:
df['PROPULSION'] = 'Combustión'
df.loc[df['CILINDRADA'] == 0, 'PROPULSION'] = 'Eléctrica'


In [31]:
# Ver todas las motos con cilindrada cero y su tipo actual
electricas = df[df['CILINDRADA'] == 0][['MARCA', 'MODELO', 'TIPO_SIMPLIFICADO', 'PROPULSION']]
print(electricas.sort_values('TIPO_SIMPLIFICADO'))


               MARCA                     MODELO TIPO_SIMPLIFICADO PROPULSION
262         KAWASAKI                      Z e-1             Naked  Eléctrica
534             VOGE                       ER10             Naked  Eléctrica
43               BMW                      CE 02           Scooter  Eléctrica
44               BMW                      CE 04           Scooter  Eléctrica
160  HARLEY DAVIDSON               LiveWire One           Scooter  Eléctrica
161  HARLEY DAVIDSON                 S2 Del Mar           Scooter  Eléctrica
162  HARLEY DAVIDSON              S2 Mulholland           Scooter  Eléctrica
163            HONDA                      EM1 e           Scooter  Eléctrica
373          PEUGEOT            E-Streetzone L1           Scooter  Eléctrica
399          PIAGGIO      Piaggio 1 electric 45           Scooter  Eléctrica
400          PIAGGIO    Piaggio 1 + electric 45           Scooter  Eléctrica
401          PIAGGIO  Piaggio 1 Activ. elec. 60           Scooter  Eléctrica

In [32]:
# LiveWire One → Naked
df.loc[df['MODELO'].str.contains("LiveWire One", case=False), 'TIPO_SIMPLIFICADO'] = 'Naked'

# S2 Del Mar → Naked
df.loc[df['MODELO'].str.contains("S2 Del Mar", case=False), 'TIPO_SIMPLIFICADO'] = 'Naked'

# S2 Mulholland → Custom (es estilo cruiser eléctrica)
df.loc[df['MODELO'].str.contains("S2 Mulholland", case=False), 'TIPO_SIMPLIFICADO'] = 'Custom'


In [33]:
df.to_csv('motofit_v3.csv', index=False)

In [34]:
#ya debería tener esto en una función.
tabla_resumen = df.pivot_table(
    values=cols_numericas,
    index='TIPO_SIMPLIFICADO',
    aggfunc='mean'
).round(1)


print("=== Tabla Resumen Técnica por Tipo de Moto ===")
print(tabla_resumen)

=== Tabla Resumen Técnica por Tipo de Moto ===
                   ALTURA_ASIENTO  CILINDRADA  PESO_VACIO  POTENCIA   PRECIO
TIPO_SIMPLIFICADO                                                           
Adventure                   828.7       793.1       244.6      88.6  13085.2
Custom                      738.3      1089.3       254.2      70.5  16542.2
Naked                       797.3       626.6       186.3      73.1   9841.7
Off-road                    895.0       408.2       139.8      44.4   8224.0
Otro                        770.3       124.3       104.3       9.6   4313.3
Scooter                     779.4       179.2       152.0      16.5   4930.0
Sport                       832.8       716.3       189.6     116.8  17611.9
Tourer                      763.6      1278.8       301.0      99.9  24175.0


In [ ]:
# He probado con diferentes características, el dataset se ve bastante correcto.

# Cargar dataset
df = pd.read_csv('motofit_v3.csv')

# Asegurar numérico
df['POTENCIA'] = pd.to_numeric(df['POTENCIA'], errors='coerce')

# Calcular IQR para la columna
Q1 = df['POTENCIA'].quantile(0.25)
Q3 = df['POTENCIA'].quantile(0.75)
IQR = Q3 - Q1

# Definir límites superior e inferior
limite_bajo = Q1 - 1.5 * IQR
limite_alto = Q3 + 1.5 * IQR

# Filtrar outliers
outliers_peso = df[(df['POTENCIA'] < limite_bajo) | (df['POTENCIA'] > limite_alto)]

# Mostrar posibles errores
print(outliers_peso[['MARCA', 'MODELO', 'POTENCIA', 'TIPO_SIMPLIFICADO']])


        MARCA     MODELO  POTENCIA TIPO_SIMPLIFICADO
275  KAWASAKI  Ninja H2R     310.0             Sport


Rocket 3 de Triumph CILINDRADA: 2458 cc . La moto con el motor más grande de producción en serie.
Kawasaki Ninja H2R POTENCIA: ¡310 CV! También real. Moto de circuito, sobrealimentada.
BMW K 1600 (varias versiones) PESO: >350 kg o Harley's se cerca de 500 kilos! Todas reales.
 
Conclusión: no son errores, pero sí posibles outliers válido.

In [38]:
import pandas as pd

# Cargar dataset actualizado
df = pd.read_csv('motofit_v3.csv')

# 1. Forma general
print("=== Forma del dataset ===")
print(f"Filas: {df.shape[0]}, Columnas: {df.shape[1]}\n")

# 2. Nombres de columnas
print("=== Columnas ===")
print(df.columns.tolist(), "\n")

# 3. Tipos de datos
print("=== Tipos de datos ===")
print(df.dtypes, "\n")

# 4. Valores nulos
print("=== Valores nulos por columna ===")
print(df.isnull().sum(), "\n")

# 5. Descripción estadística de variables numéricas
print("=== Estadísticas descriptivas ===")
print(df.describe().round(2), "\n")

# 6. Conteo de categorías
print("=== Tipos de moto ===")
print(df['TIPO_SIMPLIFICADO'].value_counts(), "\n")

print("=== Tipos de propulsión ===")
print(df['PROPULSION'].value_counts())


=== Forma del dataset ===
Filas: 598, Columnas: 16

=== Columnas ===
['MODELO', 'CON_CARNET', 'CILINDRADA', 'POTENCIA', 'RUEDA_DELANTERA', 'RUEDA_TRASERA', 'FRENO_DELANTERO', 'FRENO_TRASERO', 'ALTURA_ASIENTO', 'PESO_VACIO', 'PRECIO', 'MARCA', 'CARNET_MINIMO', 'TIPO_CORREGIDO', 'TIPO_SIMPLIFICADO', 'PROPULSION'] 

=== Tipos de datos ===
MODELO                object
CON_CARNET            object
CILINDRADA           float64
POTENCIA             float64
RUEDA_DELANTERA       object
RUEDA_TRASERA         object
FRENO_DELANTERO       object
FRENO_TRASERO         object
ALTURA_ASIENTO       float64
PESO_VACIO           float64
PRECIO               float64
MARCA                 object
CARNET_MINIMO         object
TIPO_CORREGIDO        object
TIPO_SIMPLIFICADO     object
PROPULSION            object
dtype: object 

=== Valores nulos por columna ===
MODELO                0
CON_CARNET            0
CILINDRADA            0
POTENCIA             34
RUEDA_DELANTERA      12
RUEDA_TRASERA        12
FREN

In [ ]:
motos_sin_precio = df[df['PRECIO'].isnull()][['MARCA', 'MODELO', 'TIPO_SIMPLIFICADO', 'PRECIO']]
print(motos_sin_precio)

#comprobado, no disponibles realmente.

         MARCA                 MODELO TIPO_SIMPLIFICADO  PRECIO
212  HUSQVARNA           Vitpilen 801             Naked     NaN
215  HUSQVARNA             Norden 901         Adventure     NaN
267   KAWASAKI                   Z H2             Naked     NaN
268   KAWASAKI                Z H2 SE             Naked     NaN
301        KTM      450 Rally Replica         Adventure     NaN
305        KTM          790 Adventure         Adventure     NaN
313        KTM     1390 Super Duke GT             Sport     NaN
315        KTM  1390 Super Adv. S Evo         Adventure     NaN
597     ZONTES                  703RR             Sport     NaN
